tests/zenko_tests/node_tests/backbeat/IngestionUtility.js

```diff
@@ -0,0 +1,170 @@
+const assert = require('assert');
+const async = require('async');
+const crypto = require('crypto');
+
+const { scalityS3Client, ringS3Client } = require('../s3SDK');
+
+class IngestionUtility {
+    constructor(s3, ringS3C) {
+        this.s3 = s3;
+        this.ringS3C = ringS3C;
+    }
+
+    _compareObjectBody(body1, body2) {
+        const digest1 = crypto.createHash('md5').update(body1).digest('hex');
+        const digest2 = crypto.createHash('md5').update(body2).digest('hex');
+        assert.strictEqual(digest1, digest2);
+    }
+
+
+    _deleteVersionList(versionList, bucketName, cb) {
+        async.each(versionList, (versionInfo, next) =>
+            this.deleteObject(bucketName, versionInfo.Key,
+                versionInfo.VersionId, next), cb);
+    }
+
+    putObject(bucketName, objectName, content, cb) {
+        this.s3.putObject({
+            Bucket: bucketName,
+            Key: objectName,
+            Body: content,
+        }, cb);
+    }
+
+    deleteObject(bucketName, key, versionId, cb) {
+        this.s3.deleteObject({
+            Bucket: bucketName,
+            Key: key,
+            VersionId: versionId,
+        }, cb);
+    }
+
+    getSourceObject(bucketName, objName, cb) {
+        this.ringS3C.getObject({
+            Bucket: bucketName,
+            Key: objName,
+        }, cb);
+    }
+
+    getDestObject(bucketName, objName, cb) {
+        this.s3.getObject({
+            Bucket: bucketName,
+            Key: objName,
+        }, cb);
+    }
+
+    createIngestionBucket(bucketName, ingestionSrcLocation, cb) {
+        async.series([
+            next => this.s3.createBucket({
+                Bucket: bucketName,
+                CreateBucketConfiguration: {
+                    LocationConstraint: `${ingestionSrcLocation}:ingest`,
+                },
+            }, next),
+            next => this.s3.putBucketVersioning({
```

Comment:
You don't have to do this, creating an ingestion bucket automatically enables versioning on the bucket now. These were the changes I made this week.

Reply:
Right, this was put in before those changes were merged. will remove them afterwards